src/hotspot/share/gc/g1/g1_globals.hpp

```diff
@@ -324,6 +324,10 @@
           "retained region restore purposes.")                              \
           range(1, 256)                                                     \
                                                                             \
+  product(uint, G1NumCollectionsKeepPinned, 8, DIAGNOSTIC,                  \
```

Comment:
Any particular reason this is not `EXPERIMENTAL`?

Reply:
Changing this does not in any way enable risky/experimental code not fit for production. This knob is for helping diagnose performance issues.

G1 does have its fair share of experimental options, but all/most of these were from the initial import where G1 as a whole had been experimental (unstable) for some time.